turbopack/crates/turbopack-ecmascript/src/references/mod.rs

```diff
@@ -599,7 +605,12 @@ pub(crate) async fn analyse_ecmascript_module_internal(
                     }
                     ImportedSymbol::Symbol(name) => Some(ModulePart::export((&**name).into())),
                     ImportedSymbol::PartEvaluation(part_id) => {
-                        evaluation_references.push(i);
+                        // If side effect free, ImportedSymbol::PartEvaluation is also side effect
+                        // free because ImportedSymbol::PartEvaluation is used only for
+                        // __TURBOPACK_PART__ imports so it doesn't need to be evaluated
+                        if !is_side_effect_free {
```

Comment:
This should be handled by `turbopack/crates/turbopack/src/lib.rs` where there is already code for returning `ProcessResult::Ignore` for side effect free evaluation imports